test/test_unary_ufuncs.py

```diff
@@ -93,26 +120,43 @@ def generate_numeric_tensors(device, dtype, *,
                    make_tensor(large_size, device=device, dtype=dtype, low=None, high=None))
         return tensors
 
+    low = domain[0] if domain[0] is not None else -float('inf')
+    high = domain[1] if domain[1] is not None else float('inf')
+
     # Acquires dtype-specific vals
     if dtype.is_floating_point or dtype.is_complex:
-        large_vals = _large_float_vals if include_large_values else tuple()
-        extremals = _float_extremals if include_extremal_values else tuple()
-        vals = _float_vals + large_vals + extremals
-
-        # Converts float -> complex vals if dtype is complex
+        # NOTE: large and extremal complex values are not tested
+        large_vals = _large_float_vals if include_large_values and dtype.is_floating_point else tuple()
+        opt_inf_vals = (-float('inf'), float('inf')) if dtype.is_floating_point else tuple()
+        opt_nan = (float('nan'),) if dtype.is_floating_point else tuple()
+
+        # NOTE: float (and complex) filter is strictly less than / greater than
+        #   since float domains exclude their endpoints
+        unfiltered_vals = _float_vals + large_vals + opt_inf_vals
+        filtered_vals = tuple(val for val in unfiltered_vals if (val > low and val < high))
```

Comment:
seems wasteful to create opt_inf_vals only to filter them out.

Reply:
True, although I think the performance impact will be small. Not many operators define a domain so they're typically unfiltered.

Reply:
If we really wanted to be performant we could develop a fast path for when no domain is specified.

Reply:
Ensure inf values are included when no domain is specified.

Reply:
at caller include_large_values=False for complex and True for other things

Reply:
make helper for setting up domain and filtering values

Reply:
always generate in float, next after the domain, clamp